apps/arena/native/physics/src/collision_detection/ear_clipping.rs

```diff
@@ -0,0 +1,197 @@
+2. Once we find an ear do the following:
+   - Save the triangle formed by the current previous and next vertex
+   - Delete the current vertex from the available vertex list
+   - Start again with the cleaned up vertex list until the list length is 3
+3. Make a final triangle with the 3 remaining vertices
+
+Even tho the resolution is different the logic is the same as the following article:
+- [Geometric tools article](https://www.geometrictools.com/Documentation/TriangulationByEarClipping.pdf)
+
+*/
+
+pub(crate) fn maybe_triangulate_polygon(mut polygon: Entity) -> Vec<Entity> {
+    clean_extra_vertices(&mut polygon);
+    if is_polygon_concave(&polygon) {
+        return vec![polygon.clone()];
+    }
+    triangulate_polygon(&polygon)
+}
+
+// Here we'll remove any vertices are shouldn't be part of the polygon or are useless
+// - If three vertices are in a line that's a useless connections that we should remove
+fn clean_extra_vertices(polygon: &mut Entity) {
+    let mut cleaned_vertices = vec![];
+    for current_vertex_index in 0..polygon.vertices.len() {
+        let previous_vertex_index =
+            get_previous_cyclic_index(current_vertex_index, &polygon.vertices);
+        let next_vertex_index = get_next_cyclic_index(current_vertex_index, &polygon.vertices);
+
+        let previous_vertex = polygon.vertices[previous_vertex_index];
+        let current_vertex = polygon.vertices[current_vertex_index];
+        let next_vertex = polygon.vertices[next_vertex_index];
+
+        if !is_vertex_consecutive(&previous_vertex, &current_vertex, &next_vertex) {
+            cleaned_vertices.push(current_vertex);
+        }
+    }
+
+    polygon.vertices = cleaned_vertices;
+}
+
+fn is_vertex_consecutive(
+    previous_vertex: &Position,
+    current_vertex: &Position,
+    next_vertex: &Position,
+) -> bool {
+    let first_vector = Position::sub(previous_vertex, current_vertex);
+    let second_vector = Position::sub(next_vertex, current_vertex);
+
+    get_cross_product_scalar(&first_vector, &second_vector) == 0.0
+}
+
+fn triangulate_polygon(polygon: &Entity) -> Vec<Entity> {
+    let mut result: Vec<Entity> = vec![];
+    let mut positions = polygon.vertices.clone();
+    let mut ear_found: bool;
+
+    while positions.len() > 3 {
+        ear_found = false;
+        for current_vertex_index in 0..positions.len() {
+            let previous_vertex_index = get_previous_cyclic_index(current_vertex_index, &positions);
+            let next_vertex_index = get_next_cyclic_index(current_vertex_index, &positions);
+
+            let previous_vertex = positions[previous_vertex_index];
+            let current_vertex = positions[current_vertex_index];
+            let next_vertex = positions[next_vertex_index];
+
+            if is_position_ear(current_vertex, previous_vertex, next_vertex, &positions) {
+                ear_found = true;
+                let triangle_vertex = vec![current_vertex, next_vertex, previous_vertex];
```

Comment:
This a triangle, not a vertex.
```suggestion
                let ear_triangle = vec![current_vertex, next_vertex, previous_vertex];
```

Reply:
done